packages/finance/src/price/mod.rs

```diff
@@ -66,23 +66,45 @@ where
 {
     #[track_caller]
     fn new(amount: Coin<C>, amount_quote: Coin<QuoteC>) -> Self {
-        let (amount_normalized, amount_quote_normalized) = amount.into_coprime_with(amount_quote);
-        let res = Self {
+        let res: Self = Self::new_inner(amount, amount_quote);
+
+        #[cfg(debug_assertions)]
+        res.invariant_held().unwrap();
+
+        res
+    }
+
+    #[cfg(feature = "testing")]
+    pub const fn unchecked(amount: Coin<C>, amount_quote: Coin<QuoteC>) -> Self {
+        Self::new_inner(amount, amount_quote)
+    }
```

Comment:
there is almost the same alternative, why have you added this?

Reply:
`invariant_held` can not be ran in compile time without further refactoring, thus the need for this method.